src/app/view/perspectives/kanban/util/kanban-converter.ts

```diff
@@ -0,0 +1,262 @@
+      documentsIdsOrder: otherDocumentsIds,
+    };
+    return {...config, columns, otherColumn};
+  }
+
+  private groupDocumentsByColumns(
+    documentsByCollection: Record<string, DocumentModel[]>,
+    collectionsConfig: Record<string, KanbanCollectionConfig>,
+    collections: Collection[],
+    constraintData?: ConstraintData
+  ): {columnsMap: Record<string, KanbanColumnData>; otherDocumentsIds: string[]} {
+    const columnsMap: Record<string, KanbanColumnData> = {};
+    const otherDocumentsIds = [];
+    for (const collection of collections || []) {
+      const collectionConfig = collectionsConfig[collection.id];
+      const kanbanAttribute = collectionConfig && collectionConfig.attribute && collectionConfig.attribute;
+      const attribute =
+        kanbanAttribute && (collection.attributes || []).find(attr => attr.id === kanbanAttribute.attributeId);
+      const documents = documentsByCollection[collection.id] || [];
+      if (attribute) {
+        for (const document of documents) {
+          const value = document.data[attribute.id];
+          const formattedValue = this.formatKanbanColumnValue(
+            value,
+            attribute.constraint,
+            constraintData,
+            kanbanAttribute
+          );
+          if (isNotNullOrUndefined(formattedValue) && formattedValue !== '') {
+            const createdByAttribute = {collectionId: collection.id, attributeId: attribute.id};
+            const stringValue = formattedValue.toString();
+
+            if (columnsMap[stringValue]) {
+              const columnData = columnsMap[stringValue];
+              columnData.documentsIds.push(document.id);
+              if (!kanbanAttributesIncludesAttribute(columnData.attributes, createdByAttribute)) {
+                columnData.attributes.push(createdByAttribute);
+              }
+              if (
+                attribute &&
+                attribute.constraint &&
+                !columnData.constraintTypes.includes(attribute.constraint.type)
+              ) {
+                columnData.constraintTypes.push(attribute.constraint.type);
+              }
+            } else {
+              const constraintTypes = (attribute.constraint && [attribute.constraint.type]) || [];
+              columnsMap[stringValue] = {
+                documentsIds: [document.id],
+                attributes: [createdByAttribute],
+                constraintTypes,
+              };
+            }
+          } else {
+            otherDocumentsIds.push(document.id);
+          }
+        }
+      } else {
+        otherDocumentsIds.push(...documents.map(document => document.id));
+      }
+    }
+
+    return {columnsMap, otherDocumentsIds};
+  }
+
+  private formatKanbanColumnValue(
+    value: any,
+    constraint: Constraint,
+    constraintData: ConstraintData,
+    kanbanAttribute: KanbanAttribute
+  ): any {
+    if (constraint) {
+      if (constraint.type === ConstraintType.User) {
+        return value;
```

Comment:
Is this shown as column title? Don't we want to display user name if it is available? What about using `formatUserDataValue` function?

Reply:
Yes we do. Look at `kanban-column-header.component`. In order to show user with avatar we need to format it directly in template.